src/coreclr/jit/instr.cpp

```diff
@@ -2185,6 +2185,9 @@ instruction CodeGenInterface::ins_Load(var_types srcType, bool aligned /*=false*
             case TYP_INT:
                 ins = INS_i32_load;
                 break;
+            case TYP_LONG:
```

Comment:
The way this function is currently structured is leaving performance on the table. It would be best to remove this and lower part that is dealing with FP and integers separately, and have one switch for both at the beginning.